File: app/src/main/java/com/practice/bmi/MainActivity.java
package com.practice.bmi;

import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;
import androidx.appcompat.widget.AppCompatEditText;
import androidx.core.content.res.ResourcesCompat;
import androidx.recyclerview.widget.GridLayoutManager;
import com.practice.bmi.databinding.ActivityMainBinding;
import java.util.ArrayList;
import java.util.Objects;
import www.sanju.motiontoast.MotionToast;
import www.sanju.motiontoast.MotionToastStyle;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private ActivityMainBinding binding;
    private NoteDataBaseHelper noteDataBaseHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        noteDataBaseHelper = NoteDataBaseHelper.getDB(this);
        binding.fabAdd.setOnClickListener(this);

        showNotes();
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.fabAdd) {
            Dialog dialog = new Dialog(this);
            dialog.setContentView(R.layout.layout_add_notes);
            AppCompatEditText etTitle, etDesc;
            AppCompatButton btnAdd;
            etTitle = dialog.findViewById(R.id.tvNoteTitle);
            etDesc = dialog.findViewById(R.id.tvNoteDesc);
            btnAdd = dialog.findViewById(R.id.btnAdd);
            btnAdd.setOnClickListener(v1 -> {
                String title = Objects.requireNonNull(etTitle.getText()).toString();
                String desc = Objects.requireNonNull(etDesc.getText()).toString();
                if (!title.isEmpty() && !desc.isEmpty()) {
                    noteDataBaseHelper.notesDao().addNotes(new NotesData(title, desc));
                    showNotes();

                    dialog.dismiss();
                } else {
                    MotionToast.Companion.createColorToast(this, "", "Please enter all fields!!", MotionToastStyle.ERROR,
                            MotionToast.GRAVITY_BOTTOM, MotionToast.SHORT_DURATION, ResourcesCompat.getFont(this, R.font.atkinson_medium));
                }
            });

            dialog.show();
        }
    }

    public void showNotes() {
        ArrayList<NotesData> arryNotes = (ArrayList<NotesData>) noteDataBaseHelper.notesDao().getNotes();
        if (arryNotes.size() > 0) {
            binding.rvNotes.setVisibility(View.VISIBLE);
            binding.tvNoNotes.setVisibility(View.GONE);

            binding.rvNotes.setLayoutManager(new GridLayoutManager(this, 2));
            binding.rvNotes.setAdapter(new NotesAdapter(this, arryNotes, noteDataBaseHelper));
        } else {
            binding.rvNotes.setVisibility(View.GONE);
            binding.tvNoNotes.setVisibility(View.VISIBLE);
        }
    }
}

File: app/src/main/java/com/practice/bmi/MainActivity.kt
package com.practice.bmi

import android.Manifest.permission.ACCESS_COARSE_LOCATION
import android.Manifest.permission.ACCESS_FINE_LOCATION
import android.Manifest.permission.READ_SMS
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.res.ResourcesCompat
import com.practice.bmi.databinding.ActivityMainBinding
import www.sanju.motiontoast.MotionToast
import www.sanju.motiontoast.MotionToastStyle

class MainActivity : AppCompatActivity(), View.OnClickListener {

    companion object {
      const val REQ_PERMISSION = 1
    }

    private val binding: ActivityMainBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        binding.btnRequest.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v!!.id) {
            R.id.btnRequest -> {
               if(checkPermission()){
                 //Toast.makeText(this, "Permission Already Granted!!", Toast.LENGTH_SHORT).show()
                 MotionToast.createColorToast(this, "","Permission Already Granted!!", MotionToastStyle.SUCCESS, MotionToast.GRAVITY_BOTTOM, MotionToast.SHORT_DURATION, ResourcesCompat.getFont(this, R.font.atkinson_medium))
               }else{
                 ActivityCompat.requestPermissions(this, arrayOf(READ_SMS, ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION), REQ_PERMISSION)
               }
            }
        }
    }

    private fun checkPermission() : Boolean {
      val permissions = arrayOf(READ_SMS, ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION)
      return permissions.all { permission ->
        ActivityCompat.checkSelfPermission(this, permission) == PackageManager.PERMISSION_GRANTED
      }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>,
                                            grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)

        if(requestCode == REQ_PERMISSION){
          if(grantResults.isNotEmpty()){
            val sms = grantResults[0]
            val fineLoc = grantResults[1]
            val coarseLoc = grantResults[2]

            val checkSms = sms == PackageManager.PERMISSION_GRANTED
            val checkFL = fineLoc == PackageManager.PERMISSION_GRANTED
            val checkCL = coarseLoc == PackageManager.PERMISSION_GRANTED
              
             if(checkSms && checkFL && checkCL){
                 //Toast.makeText(this, "Permissions Granted!!", Toast.LENGTH_SHORT).show()
                 MotionToast.createColorToast(this, "","Permission Granted!!", MotionToastStyle.SUCCESS, MotionToast.GRAVITY_BOTTOM, MotionToast.SHORT_DURATION, ResourcesCompat.getFont(this, R.font.atkinson_medium))
             }else {
                 //Toast.makeText(this, "Permissions Denied!!", Toast.LENGTH_SHORT).show()
                 MotionToast.createToast(this, "", "Permission Denied!!", MotionToastStyle.WARNING, MotionToast.GRAVITY_BOTTOM, MotionToast.SHORT_DURATION, ResourcesCompat.getFont(this, R.font.atkinson_medium))
             } 
          }
        }
    }
}